src/state_point.F90

```diff
@@ -214,18 +218,55 @@ subroutine write_state_point()
              group="tallies/tally" // to_str(i), length=t % n_nuclide_bins)
         deallocate(temp_array)
 
-        ! Write number of score bins, score bins, and moment order
+        ! Write number of score bins, score bins, user score bins
         call sp % write_data(t % n_score_bins, "n_score_bins", &
              group="tallies/tally" // to_str(i))
         call sp % write_data(t % score_bins, "score_bins", &
              group="tallies/tally" // to_str(i), length=t % n_score_bins)
-        call sp % write_data(t % moment_order, "moment_order", &
-             group="tallies/tally" // to_str(i), length=t % n_score_bins)
-
-        ! Write number of user score bins
         call sp % write_data(t % n_user_score_bins, "n_user_score_bins", &
              group="tallies/tally" // to_str(i))
 
+      ! Write explicit moment order strings for each score bin
+#ifdef HDF5
+        call sp % open_group("tallies/tally" // trim(to_str(i)) // "/moments")
+        call sp % close_group()
+#endif
```

Comment:
Can this #ifdef block be removed now? If I'm not mistaken, the call to `sp % write_data` will automatically create the group if it doesn't exist.